extensions/ql-vscode/src/queries.ts

```diff
@@ -173,11 +172,12 @@ export class QueryInfo {
 /**
  * Call cli command to interpret results.
  */
-export async function interpretResults(server: cli.CodeQLCliServer, queryInfo: QueryInfo, resultsInfo: ResultsInfo, sourceInfo?: cli.SourceInfo): Promise<sarif.Log> {
-  if (await fs.pathExists(resultsInfo.interpretedResultsPath)) {
-    return JSON.parse(await fs.readFile(resultsInfo.interpretedResultsPath, 'utf8'));
+export async function interpretResults(server: cli.CodeQLCliServer, metadata: QueryMetadata | undefined, resultsPath: string, sourceInfo?: cli.SourceInfo): Promise<sarif.Log> {
+  const interpretedResultsPath = resultsPath + ".interpreted.sarif"
```

Comment:
Was there something wrong about computing all the requisite paths for a query up front? Is there a situation in which we need the flexibility of multiple different results files for a single `queryInfo`, necessitating passing the `resultsPath` down by itself? I was trying to isolate all the filename generation to a single place.

Reply:
Ok I have switched back to generating them in one place. I had to change some things around to pass the data through the UI. So this may need a fairly large re-review.